Adds tracking for all of the site settings areas

emilylaguna opened the pull request:
Project: #17503

⚠️ Depends on: https://github.com/wordpress-mobile/WordPress-iOS/pull/17539
View only this PRs [changes here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17545/files/b1803c087a1f29c4a8d24116579f02ad999eb58c..87a15d908ab723ccfda37652e5abb3224a41f8e7)


### Description
Adds tracking for **every** option in Site Settings and its encompassing settings

### To test:
This affects every item on Site Settings that did not already have tracks

1. Launch the app
2. Tap on the My Site tab
3. Tap on Site Settings
4. Go through each setting on the main view, and the sub views and verify you see for each of the settings changed:
    - `🔵 Tracked: settings_did_change <field_name: FIELD_NAME, page: PAGE_NAME>`

Note: Areas with existing tracks were not changed, and tracking was not added to those areas

### Regression Notes
1. Potential unintended areas of impact
None, adding tracking.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
None.

### PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Add new settings track names

Commit:
Add a fieldName property to EditableTextRow to help in tracking which field is being changed

Commit:
Add a trackingKey property to the SettingsController's

Commit:
Add a helper method to track when a SettingsController's text field is changed

Commit:
Add fieldName's to the Account Settings view

Commit:
Track when the primary site is changed

Commit:
Add fieldName's to the My Profile settings view

Commit:
Remove account_close_cancel event

Commit:
Add tracking when the user closes their account

Commit:
Track when the clear media cache button is tapped

Commit:
Track when the app settings clear spotlight index button is tapped

Commit:
Track when the clear app settings siri suggestions button is tapped

Commit:
Track the app icon name when the app icon is changed

Commit:
Add track name for the privacy settings

Commit:
Track when the privacy settings is opened and the report crashes option is toggled

Commit:
Add tracking when the open device settings button is tapped

Commit:
Add tracking to the site settings view

Commit:
Track homepage settings change

Commit:
Track default category change

Commit:
Add a helper method for tracking settings changes

Commit:
Update existing references to use new helper method

Commit:
Add edit/add tag tracking to site settings

Commit:
Add objc support to the trackSettingsChange helper method

Commit:
Add tracking to the date and time format settings view

Commit:
Add tracking to the related posts settings view

Commit:
Add default post format tracking

Commit:
Add speed up your site tracking

Commit:
Add discussion settings tracking

Commit:
Add tracking to jetpack settings

momo-ozawa reviewed (commented):
@emilylaguna I didn't see any tracking for the Privacy and Language settings under Site Settings
<img width="377" alt="Screen Shot 2021-11-24 at 11 00 27" src="https://user-images.githubusercontent.com/6711616/143226334-ae196928-69b0-49e7-b118-f90dde1c6af1.png">

frosty commented on WordPress/Classes/ViewRelated/Blog/Site Settings/DiscussionSettingsViewController.swift:
```diff
@@ -268,6 +277,7 @@ open class DiscussionSettingsViewController: UITableViewController {
             }
 
             self?.settings.commentsThreading = newThreadingDepth
+            WPAnalytics.trackSettingsChange("site_settings_discussion", fieldName: "comments_threading", value: selected as Any)
```
I think most of these settings keys should be stored as constants somewhere (and likely field names), as the same literals are reproduced in many places and a typo could mean we end up tracking the wrong thing. This comment applies to basically every `trackSettingsChange` call 🙂

emilylaguna replied:
I can see this point, but all of them are stored as private constants in [BlogServiceRemoteREST](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/c5e11d569989e6782330200121545832c2a3b3d5/WordPressKit/BlogServiceRemoteREST.m) and it feels like overkill to expose these constants, or duplicate them in a public fashion.

frosty replied:
Hmm, I guess I mean if nothing else we could have a constant in `DiscussionSettingsViewController` for `let tracksDiscussionSettingsKey = "site_settings_discussion"` and so forth?

emilylaguna replied:
I refactored the tracking a bit more on that view, added a constant, and moved all of the tracking to a helper func.

frosty reviewed (changes requested):
Hello! As Momo mentioned, I also don't see tracking for Privacy or Language.

Under Discussion, the events for Sending and Receiving pingbacks seem to be reversed (toggling sending pingbacks fires the 'receive' event and vice versa).

I also added a code level comment as I feel most of these tracking calls should use constant values for the events.

Commit:
Adds tracking for the amp, privacy, and language fields in site settings

Commit:
Swap send and receive pingbacks tracking

emilylaguna commented:
> @emilylaguna I didn't see any tracking for the Privacy and Language settings under Site Settings <img alt="Screen Shot 2021-11-24 at 11 00 27" width="377" src="https://user-images.githubusercontent.com/6711616/143226334-ae196928-69b0-49e7-b118-f90dde1c6af1.png">

Oops, my sites didn't have this appear. Fixed now, and added tracking for some other options as well.

Commit:
Refactor the discussion settings tracking

frosty reviewed (approved):
:shipit:
